### PR TITLE
Add lint and smoke tests

### PR DESCRIPTION
**Pull request by jardon**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
No tests are run on PR

* **What is the new behavior (if this is a feature change)?**
Lint and smoke tests are ran on PR event.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Nope

* **Other information**:
N/A

### CODE REVIEW

**Review by MiaAltieri — COMMENTED** [review]
Excellent smoke tests. One knit is I think it might be nice to have a generalised `ci.yaml` file like we do in the [MongoDB vm charm](https://github.com/canonical/mongodb-operator/blob/main/.github/workflows/ci.yaml) and k8s charm. this way things will be consistent across products - thoughts?

**Comment by jardon** [comment]
> Excellent smoke tests. One knit is I think it might be nice to have a generalised `ci.yaml` file like we do in the [MongoDB vm charm](https://github.com/canonical/mongodb-operator/blob/main/.github/workflows/ci.yaml) and k8s charm. this way things will be consistent across products - thoughts?

I used to like writing workflows like this, but I ultimately stopped because its really an antipattern.  Generally, I'd agree that its good to be consistent so I can update the workflows if thats what is requested, but I don't see really any added value otherwise.  I could make `ci.yaml` that just calls all the other workflows and remove the `pull_request` triggers from the other workflows if thats helpful at all.

**Comment by MiaAltieri** [comment]
Noob question why is in an "antipattern"?

**Review by MiaAltieri — APPROVED** [review]
Thanks for the quality explainaition on this today! I like the pattern and it makes sense to me :)

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install tox
+        run: python3 -m pip install tox
+      - name: YAML Lint
+        run: tox -e lint
```
Comment (carlcsaposs-canonical) [inline .github/workflows/lint.yaml]:
```suggestion
        run: tox run -e lint
```

File: .github/workflows/smoke.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install tox
+        run: python3 -m pip install tox
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - name: Smoke Tests
+        run: tox -e smoke
```
Comment (carlcsaposs-canonical) [inline .github/workflows/smoke.yaml]:
```suggestion
        run: tox run -e smoke
```

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[tox]
+requires =
+	tox>=4
```
Comment (carlcsaposs-canonical) [inline tox.ini]: FYI if tox v4 is not installed this will install tox inside tox which might make debugging virtual env issues more confusing

Comment (carlcsaposs-canonical) [inline tox.ini]:
```suggestion
requires =
    tox>=4
```
4 space instead of tab indent

**Review by carlcsaposs-canonical — APPROVED** [review]
nits, feel free to ignore

**Review by carlcsaposs-canonical — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]
excellent work

**Pull request merged**